src/App.tsx

```diff
@@ -36,12 +36,28 @@ const App = (): JSX.Element => {
     const addPurchase = (purchase: IPurchase) => {
         setPurchaseList([...purchaseList, purchase])
     }
+    const editPurchase = (purchase: IPurchase) => {
+        purchase.isEdit = !purchase.isEdit
+        setPurchaseList([...purchaseList])
+    }
+    const deletePurchase = (purchaseId: number) => {
+        purchaseList.forEach((purchase, index) => {
+            if (purchaseId === purchase.id) {
+                purchaseList.splice(index, 1)
+            }
+        })
+        setPurchaseList([...purchaseList])
+    }
```

Comment:
Don't iterate over all array just to find the right element in it. Use findIndex to find the correct element and then do slice (not splice, don't modify old data, create new data depending on the old data).